test/semantics/modfile07.f90

```diff
@@ -0,0 +1,69 @@
+! Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+!
+! Licensed under the Apache License, Version 2.0 (the "License");
+! you may not use this file except in compliance with the License.
+! You may obtain a copy of the License at
+!
+!     http://www.apache.org/licenses/LICENSE-2.0
+!
+! Unless required by applicable law or agreed to in writing, software
+! distributed under the License is distributed on an "AS IS" BASIS,
+! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+! See the License for the specific language governing permissions and
+! limitations under the License.
+
+! Check modfile generation for generic interfaces
+module m
+  interface foo
+    subroutine s1(x)
+      real x
+    end subroutine
+    subroutine s2(x)
+      complex x
+    end subroutine
+  end interface
+  interface bar
+    procedure :: s1
+    procedure :: s2
+    procedure :: s3
+    procedure :: s4
+  end interface
+contains
+  subroutine s3(x)
+    logical x
+  end
+  subroutine s4(x)
+    integer x
+  end
+end
+
+!Expect: m.mod
+!module m
+! interface foo
+!  procedure::s1
+!  procedure::s2
+! end interface
```

Comment:
FYI, I'm thinking about changing this generic interface to be written as:
`generic::foo=>s1,s2`
It's more compact and allows for an access-spec.

Reply:
Size savings probably don't matter much, but the `GENERIC` statement does seem to be the most general way to write such things.  I'm pretty sure that it can be used for `OPERATOR` and I/O overloads, but you might want to confirm that.